web/src/desktop_notifications.ts

```diff
@@ -0,0 +1,111 @@
+import $ from "jquery";
+import assert from "minimalistic-assert";
+
+import type {Message} from "./message_store";
+
+type NoticeMemory = Map<
+    string,
+    {
+        obj: Notification | ElectronBridgeNotification;
+        msg_count: number;
+        message_id: number;
+    }
+>;
+
+export const notice_memory: NoticeMemory = new Map();
+
+export let NotificationAPI: typeof ElectronBridgeNotification | typeof Notification;
```

Comment:
This type should include `undefined`. (I think it will indeed be `undefined` in iOS Safari < 16.4.)